crates/core/src/rpc/minimal.rs

```diff
@@ -668,12 +668,40 @@ impl Minimal for SurfpoolMinimalRpc {
     fn get_block_height(
         &self,
         meta: Self::Metadata,
-        _config: Option<RpcContextConfig>,
+        config: Option<RpcContextConfig>,
     ) -> Result<u64> {
+        let config = config.unwrap_or_default();
+
+        //get the current slot and check min_context_slot constraint
+        let current_slot = meta
+            .with_svm_reader(|svm_reader| svm_reader.get_latest_absolute_slot())
+            .map_err(Into::<jsonrpc_core::Error>::into)?;
+
+
+        // check minimum context slot whn specified -> against current slot, not adjusted slot
+        if let Some(min_context_slot) = config.min_context_slot {
+            if current_slot < min_context_slot {
+                return Err(RpcCustomError::MinContextSlotNotReached {
+                    context_slot: min_context_slot,
+                }
+                    .into());
+            }
+        }
+
+        // commitment adjustments to determine which slot's data to use
+        let _target_slot = match config.commitment.unwrap_or_default().commitment {
+            CommitmentLevel::Processed => current_slot,
+            CommitmentLevel::Confirmed => current_slot.saturating_sub(1),
+            CommitmentLevel::Finalized => current_slot.saturating_sub(FINALIZATION_SLOT_THRESHOLD),
+        };
+
+
+        // return the current block height since we don't have historical block height data in the current implementation
```

Comment:
The `SurfnetSvm` struct has `pub blocks: HashMap<Slot, BlockHeader>`, which can be used to get the block height associated with a slot